src/java.base/share/classes/java/util/zip/Deflater.java

```diff
@@ -730,41 +731,41 @@ public int deflate(ByteBuffer output, int flush) {
                 inputPos = input.position();
                 int inputRem = Math.max(input.limit() - inputPos, 0);
                 if (input.isDirect()) {
-                    long inputAddress = ((DirectBuffer) input).address();
+                    var inScope = NIO_ACCESS.acquireSession(input);
                     try {
                         if (output.isDirect()) {
-                            long outputAddress = outputPos + ((DirectBuffer) output).address();
+                            var outScope = NIO_ACCESS.acquireSession(output);
                             try {
                                 result = deflateBufferBuffer(zsRef.address(),
-                                    inputAddress + inputPos, inputRem,
-                                    outputAddress, outputRem,
+                                    ((DirectBuffer) input).address() + inputPos, inputRem,
```

Comment:
Another one where inputAddress and outputAddress have been removed so the call to deflatBufferBuffer is much hard to digest.